### ISSUE TITLE
CIncludeParser does not handle localization

### ISSUE THREAD

**Issue by poppolopoppo** [fastbuild/fastbuild#33]
Hi again,

So, after my pull request about unity time stamp "fix" (https://github.com/fastbuild/fastbuild/pull/18) I  continued to investigate my false-nothing-to-build-state while using unities. 

I am French and my compiler does not output :
````
Note: including file: X:\A\B\C
````
But :
````
Remarque : inclusion du fichier : X:\A\B\C
````
Which is not recognized by the parser, resulting in ObjectNodes without dynamic dependencies.
The problem is coming from [CIncludeParser::ParseMSCL_Preprocessed()](https://github.com/fastbuild/fastbuild/blob/master/Code/Tools/FBuild/FBuildCore/Helpers/CIncludeParser.cpp#L54), which does not handle  compiler message localization :

In the end It was not just the unities, but all ObjectNodes that were missing their includes.
Your code is handling all the dependencies nicely, It just needs a better include parser ;)

**Comment by ffulin** [comment]
There are two different include extraction paths for the MS compiler. The other one uses the preprocessor and checks for #line directives in the preprocessed output, so at least that one should be ok (since #line is a keyword and not localized).  (This path is used when compiling in distributed mode or using the cache).

The "Show Includes" path is used when we don't need to do 2-pass compilation as it's a little bit faster. Even a distributed build can have some nodes go through this path. It's unfortunate that these messages are localized, but not that surprising.

I guess the fix for this will have to be to change the parser to search for the pattern:
.*:.*: path

**Comment by poppolopoppo** [comment]
Yes the other method is not a problem. For now I just use a simple fix which greps "Remarque : inclusion du fichier :" in addition to "Note: including file:". And everything behind is now
working like a charm, I am so glad to have fix this very annoying dependency problem. 

I'll surely do a pull request with a more robust solution. Parsing `.:.: path` seems ok, but this parsing rule is pretty weak and it will be necessary to check if parsed file names exist to guard against false positives (the current parser has a strong enough parsing rule to bypass that).

**Comment by ffulin** [comment]
I think checking for the existence of the files will be too slow. Note that this is only parsing in the /showIncludes case, where there is typically very little other output. I think the chance of false positives is very low.

**Comment by poppolopoppo** [comment]
Ok, I added a [pull request](https://github.com/fastbuild/fastbuild/pull/37) with a fix (which is not checking files for existence).

**Comment by ffulin** [comment]
The fix has been integrated into dev and the Perforce mainline and will be part of the next release (v0.84).

I'll leave this issue open until the new version is released.

**Comment by ffulin** [comment]
v0.84 has now been released.  Closing this issue.

**Comment by poppolopoppo** [comment]
Alas, this solution is not working well. 
Warning messages -which are also printed on stderr- can get caught in the parser :
````
C:\XXX\header.h(XX): warning CXXX: 'XXX' : blablabla
````
Will be parsed as include :
````
warning CXXX: 'XXX' : blablabla
````
This will add an always missing dependency to the node and thus invoke compilation systematically on every call (I found the bug like that).

We have to find a way to handle localization with a more robust method.
I am currently testing a fix which tests parsed included files for existence. This might not be the best solution as you pointed out, but it has the advantage of preventing this annoying bug.

**Comment by ffulin** [comment]
I've reproduced the issue - it seems it will only happen if you run without "warnings as errors". (I.e. if warnings are treated as errors, the compilation fails and the bad dependency is never saved).

Seems like the path validation is not strict enough. I'll reproduce this in a unit test and try to make a fix.

**Comment by ffulin** [comment]
I've fixed the bug and released a new version (v0.85).
I'll continue to think if there is a better approach for this, but it should be fairly unlikely to have a similar issue since the possible output being parsed is very limited.
(Fix includes a test for this specific case to prevent regression)
